mypyc/ir/rtypes.py

```diff
@@ -571,10 +551,11 @@ def __repr__(self) -> str:
                                                           in zip(self.names, self.types)))
 
     def __eq__(self, other: object) -> bool:
-        return isinstance(other, RStruct) and self.info == other.info
+        return (isinstance(other, RStruct) and self.name == other.name
+                and self.names == other.names and self.types == other.types)
 
     def __hash__(self) -> int:
-        return hash(self.info)
+        return hash((self.name, self.names, self.types))
```

Comment:
Python doesn't let you take a hash of a list. Convert to tuples, for example (e.g. `tuple(self.names)`). Add unit test for equality and hashing.